dsn.go

```diff
@@ -253,13 +253,18 @@ func ParseDSN(dsn string) (cfg *Config, err error) {
 }
 
 func fillMissingConfigParameters(cfg *Config) error {
+	if strings.Trim(cfg.Authenticator, " ") == "" {
+		cfg.Authenticator = defaultAuthenticator
+	}
 	if strings.Trim(cfg.Account, " ") == "" {
 		return ErrEmptyAccount
 	}
 	if strings.Trim(cfg.User, " ") == "" {
 		return ErrEmptyUsername
 	}
-	if strings.Trim(cfg.Password, " ") == "" {
+	authenticator := strings.ToUpper(cfg.Authenticator)
+	if authenticator != authenticatorExternalBrowser && authenticator != authenticatorOAuth && strings.Trim(cfg.Password, " ") == "" {
```

Comment:
additional change.
`password` is not required parameter for EXTERNALBROWER and OAUTH authenticators.

Reply:
Looks good